### PR TITLE
DOC: updated `pairs.py`

### PR DESCRIPTION
**Pull request by Schefflera-Arboricola**

### CODE REVIEW

File: networkx/algorithms/assortativity/pairs.py [hunk 1]
```diff
@@ -1,26 +1,27 @@
...
 import networkx as nx
 
 __all__ = ["node_attribute_xy", "node_degree_xy"]
 
 
 @nx._dispatchable(node_attrs="attribute")
 def node_attribute_xy(G, attribute, nodes=None):
-    """Returns iterator of node-attribute pairs for all edges in G.
+    """Yields a 2-tuple of (attribute, attribute) values for a specificed
+    node-attribute for all edges in G.
```
Comment (rossbar) [inline networkx/algorithms/assortativity/pairs.py:10]:
```suggestion
    """Yields a 2-tuple of (`attribute`, `attribute`) values for a specified
    node-attribute for all edges in `G`.
```

File: networkx/algorithms/assortativity/pairs.py [hunk 2]
```diff
@@ -61,7 +63,7 @@ def node_attribute_xy(G, attribute, nodes=None):
 
 @nx._dispatchable(edge_attrs="weight")
 def node_degree_xy(G, x="out", y="in", weight=None, nodes=None):
-    """Generate node degree-degree pairs for edges in G.
+    """Yields a 2-tuple of (degree, degree) values for all edges in G.
```
Comment (rossbar) [inline networkx/algorithms/assortativity/pairs.py:66]:
```suggestion
    """Yields 2-tuples of ``(degree, degree)`` values for all edges in `G`.
```

**Review by rossbar — APPROVED** [review]
A few minor suggestions but otherwise LGTM, thanks @Schefflera-Arboricola !

File: networkx/algorithms/assortativity/pairs.py [hunk 3]
```diff
@@ -1,26 +1,27 @@
...
 import networkx as nx
 
 __all__ = ["node_attribute_xy", "node_degree_xy"]
 
 
 @nx._dispatchable(node_attrs="attribute")
 def node_attribute_xy(G, attribute, nodes=None):
-    """Returns iterator of node-attribute pairs for all edges in G.
+    """Yields a 2-tuple of (`attribute`, `attribute`) values for a specified
+    node-attribute for all edges in `G`.
```
Comment (dschult) [inline networkx/algorithms/assortativity/pairs.py:10]: Can we make this the paragraph and have a shorter one-liner description for hyperlinks in the docs?
```suggestion
    """Yields 2-tuples of node attribute values for all edges in `G`.
    
    This generator yields, for each edge in `G` incident to a node in `nodes`,
    a 2-tuple of form ``(attribute value,  attribute value)`` for the parameter
    specified node-attribute.  for all edges in `G`.
```

File: networkx/algorithms/assortativity/pairs.py [hunk 4]
```diff
@@ -61,7 +63,7 @@ def node_attribute_xy(G, attribute, nodes=None):
 
 @nx._dispatchable(edge_attrs="weight")
 def node_degree_xy(G, x="out", y="in", weight=None, nodes=None):
-    """Generate node degree-degree pairs for edges in G.
+    """Yields 2-tuples of ``(degree, degree)`` values for all edges in `G`.
```
Comment (dschult) [inline networkx/algorithms/assortativity/pairs.py:66]:
```suggestion
    """Yields 2-tuples of ``(degree, degree)`` values for edges in `G`.

    This generator yields, for each edge in `G` incident to a node in `nodes`,
    a 2-tuple of form ``(degree, degree)``. The node degrees are weighted
    when a `weight` attribute is specified.
```

**Review by dschult — COMMENTED** [review]
This looks good -- My suggestions are to back off on saying `all edges` and instead describe which edges. That goes more smoothly in a paragraph, leaving a short one-line description at the top of the doc_string.

File: networkx/algorithms/assortativity/pairs.py [hunk 5]
```diff
@@ -7,7 +7,7 @@
 @nx._dispatchable(node_attrs="attribute")
 def node_attribute_xy(G, attribute, nodes=None):
     """Yields 2-tuples of node attribute values for all edges in `G`.
-    
+
     This generator yields, for each edge in `G` incident to a node in `nodes`,
     a 2-tuple of form ``(attribute value,  attribute value)`` for the parameter
     specified node-attribute.  for all edges in `G`.
```
Comment (dschult) [inline networkx/algorithms/assortativity/pairs.py:13]: my mistake - I left a half sentence "for all edges in `G`." at the end of line 13.

File: networkx/algorithms/assortativity/pairs.py [hunk 6]
```diff
@@ -1,26 +1,30 @@
...
 
 
 @nx._dispatchable(node_attrs="attribute")
 def node_attribute_xy(G, attribute, nodes=None):
-    """Returns iterator of node-attribute pairs for all edges in G.
+    """Yields 2-tuples of node attribute values for all edges in `G`.
+
+    This generator yields, for each edge in `G` incident to a node in `nodes`,
+    a 2-tuple of form ``(attribute value,  attribute value)`` for the parameter
+    specified node-attribute.  for all edges in `G`.
```
Comment (dschult) [inline networkx/algorithms/assortativity/pairs.py:13]:
```suggestion
    specified node-attribute.
```

**Review by dschult — APPROVED** [review]
I think this PR is ready to merge. I found a type I left in one of my suggestions. So am making that suggestion here.

**Pull request merged**